trace/export.go

```diff
@@ -30,30 +30,38 @@ type Exporter interface {
 	ExportSpan(s *SpanData)
 }
 
+type exportersMap map[Exporter]struct{}
+
 var (
-	exportersMu sync.Mutex
-	exporters   map[Exporter]struct{}
+	exporters atomic.Value
 )
 
 // RegisterExporter adds to the list of Exporters that will receive sampled
 // trace spans.
 //
 // Binaries can register exporters, libraries shouldn't register exporters.
 func RegisterExporter(e Exporter) {
-	exportersMu.Lock()
-	if exporters == nil {
-		exporters = make(map[Exporter]struct{})
+	new := make(exportersMap)
+	if old, ok := exporters.Load().(exportersMap); ok {
+		for k, v := range old {
+			new[k] = v
+		}
 	}
-	exporters[e] = struct{}{}
-	exportersMu.Unlock()
+	new[e] = struct{}{}
+	exporters.Store(new)
```

Comment:
I think we still need a mutex in RegisterExporter (but not when reading) to synchronize writers. See the "ReadMostly" example in https://golang.org/pkg/sync/atomic/#Value.

Reply:
I purposely didn't do that because I didn't see a major use case for concurrent registration / deregistration of exporters. 

We have 2 choices:
1. add locks
2. document register and deregister not being concurrency safe.

Reply:
Yeah I would normally agree but since concurrent registration/deregistration won't be picked up by the race detector I would say its worth it in this case.